implement-cowsay/cow.py

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env python3
+import cowsay
+import argparse
+
+def main():
+
+    animals = cowsay.char_names
+
+    # Set up command-line argument parsing
+    parser = argparse.ArgumentParser( prog="cowsay", description="Make animals say things")
+    parser.add_argument(
+        "--animal",
+        choices=animals,
+        default="cow",
+        help="The animal to be saying things."
+    )
+    parser.add_argument(
+        "message",
+        nargs="+",
+        help="The message to say."
+    )
+
+    args = parser.parse_args()
+
+    message = " ".join(args.message)
+
+    say_function = getattr(cowsay, args.animal)
+    say_function(message)
```

Comment:
You could use `get_output_string` instead, I'd generally avoid using reflection methods like `getattr` if there are alternatives as they harm readability: https://pypi.org/project/cowsay/